feature_tests/c/include/DeprecatedEnum.d.h

```diff
@@ -0,0 +1,22 @@
+#ifndef DeprecatedEnum_D_H
+#define DeprecatedEnum_D_H
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include "diplomat_runtime.h"
+
+
+
+
+
+typedef enum DeprecatedEnum {
```

Comment:
nit: would prefer for the tests to deprecate existing types in attrs.rs where possible.

Reply:
it's annoying because those types are actually used so we'd have to `#[allow(deprecated)]` everywhere

Reply:
I'm fine with a global allow on the crate